uploadfs.js

```diff
@@ -67,14 +66,60 @@ function Uploadfs() {
 
     // Load image backend
     self._image = options.image;
-    if (typeof (self._image) === 'string') {
-      self._image = require('./lib/image/' + self._image + '.js')();
+    // Throw warnings about deprecated processors or load default
+    if (typeof self._image === 'string') {
+      if (self._image === 'imagemagick') {
+        console.error('We now recommend the use of the sharp.js library (included) for image processing.');
+      }
+      if (self._image === 'jimp' || self._image === 'imagecrunch') {
+        console.error('The specified processor is no longer supported, defaulting to the sharp.js library.');
+        self._image = 'sharp';
+      }
+    }
+
+    let fallback = false;
+    // if processor is passed as an string (including 'imagemagick' or 'sharp'), try to load it or fail with warning
+    // if undefined try to load sharp and fallback to imagemagick upon fail
+    if (typeof self._image === 'string' || self._image === undefined) {
+      self._image = self._image === undefined ? 'sharp' : self._image;
+      try {
+        self._image = require(`./lib/image/${self._image}.js`)();
+      } catch {
+        if (self._image === 'sharp') {
+          console.error('Sharp not installed. We highly recommend you run \'npm install sharp\'. Trying to fall back to imagemagick.');
```

Comment:
This does not make sense, "sharp" is already a dependency of uploadfs and so if you have this version of uploadfs and sharp can run on your computer, you will already have it at this point. Just say "Sharp not available on this operating system. Trying to fall back to imagemagick."

Reply:
Changed. I guess I don't fully get the 'optional dependencies. Is it just for the use case of allowing `npm i`/`npm ci` to not fail - rather than allowing users to slim down their load using `npm I --no-optional`?

Reply:
Optional dependencies are used when the module is still useful without them. A good example is the mongodb driver module which includes a C++ driver but falls back to a slower JavaScript driver if it can't be compiled on a particular system.

Similarly, if Sharp is not available uploadfs can still be used if imagemagick is installed on the system.

Typing "npm install sharp" would not do the developer any good because that is what just failed to happen.

Reply:
We're doing this to increase the probability that we won't be forced to bring back jimp for the sake of a minority of users. If its a very small group it's reasonable to ask them to install imagemagick. If it's a larger group we might have to make an optional, separate jimp module available.